src/test/java/it/gov/pagopa/reward/service/reward/UserInitiativeCountersUpdateServiceImplTest.java

```diff
@@ -0,0 +1,109 @@
+package it.gov.pagopa.reward.service.reward;
+
+import it.gov.pagopa.reward.dto.InitiativeConfig;
+import it.gov.pagopa.reward.dto.RewardTransactionDTO;
+import it.gov.pagopa.reward.model.counters.Counters;
+import it.gov.pagopa.reward.model.counters.InitiativeCounters;
+import it.gov.pagopa.reward.model.counters.UserInitiativeCounters;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
+
+import java.math.BigDecimal;
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+class UserInitiativeCountersUpdateServiceImplTest {
```

Comment:
let's test also the condition where for an initiative there is not a reward, so all the other initiative are updated, but not the not rewarded